"TypeError: interpreted classes cannot inherit from compiled" after get_customize_class_mro_hook

When using plugin, that provides `get_customize_class_mro_hook`, mypy errors out with `TypeError: interpreted classes cannot inherit from compiled` traceback. Tested on mypy-0.701 and python-3.6.

Consider this minimal plugin:

```python
from mypy.plugin import Plugin

class DummyPlugin(Plugin):
    def get_customize_class_mro_hook(self, fullname):
        def analyze(classdef_ctx):
            pass
        return analyze

def plugin(version):
    return DummyPlugin
```

When mypy is run with this plugin enabled on simple python file, e.g. 

```python
class Something:
    pass
```

Mypy will error out with this traceback:
```
tmp/sample.py:1: error: INTERNAL ERROR -- please report a bug at https://github.com/python/mypy/issues version: 0.701
Traceback (most recent call last):
  File "ve/bin/mypy", line 10, in <module>
    sys.exit(console_entry())
  File "mypy/semanal.py", line 3791, in accept
  File "mypy/nodes.py", line 895, in accept__Node_glue
  File "mypy/nodes.py", line 896, in accept
  File "mypy/semanal.py", line 804, in visit_class_def__StatementVisitor_glue
  File "mypy/semanal.py", line 807, in visit_class_def
  File "mypy/semanal.py", line 819, in analyze_class
  File "mypy/semanal.py", line 1229, in analyze_base_classes
  File "mypy/semanal.py", line 1250, in calculate_class_mro
TypeError: interpreted classes cannot inherit from compiled
tmp/sample.py:1: : note: use --pdb to drop into pdb
```

--- a/mypy/newsemanal/semanal.py
+++ b/mypy/newsemanal/semanal.py
@@ -74,7 +74,8 @@ from mypy.nodes import (
     PlaceholderNode, COVARIANT, CONTRAVARIANT, INVARIANT,
     nongen_builtins, get_member_expr_fullname, REVEAL_TYPE,
     REVEAL_LOCALS, is_final_node, TypedDictExpr, type_aliases_target_versions,
-    EnumCallExpr, RUNTIME_PROTOCOL_DECOS
+    EnumCallExpr, RUNTIME_PROTOCOL_DECOS,
+    FakeExpression,
 )
 from mypy.tvar_scope import TypeVarScope
 from mypy.typevars import fill_typevars
@@ -1459,7 +1460,7 @@ class NewSemanticAnalyzer(NodeVisitor[None],
         if defn.fullname:
             hook = self.plugin.get_customize_class_mro_hook(defn.fullname)
             if hook:
-                hook(ClassDefContext(defn, Expression(), self))
+                hook(ClassDefContext(defn, FakeExpression(), self))
 
     def update_metaclass(self, defn: ClassDef) -> None:
         """Lookup for special metaclass declarations, and update defn fields accordingly.

--- a/mypy/nodes.py
+++ b/mypy/nodes.py
@@ -179,6 +179,15 @@ class Expression(Node):
         raise RuntimeError('Not implemented')
 
 
+class FakeExpression(Expression):
+    """A dummy expression.
+
+    We need a dummy expression in one place, and can't instantiate Expression
+    because it is a trait and mypyc barfs.
+    """
+    pass
+
+
 # TODO:
 # Lvalue = Union['NameExpr', 'MemberExpr', 'IndexExpr', 'SuperExpr', 'StarExpr'
 #                'TupleExpr']; see #1783.

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -59,6 +59,7 @@ from mypy.nodes import (
     COVARIANT, CONTRAVARIANT, INVARIANT, UNBOUND_IMPORTED, nongen_builtins,
     get_member_expr_fullname, REVEAL_TYPE, REVEAL_LOCALS, is_final_node,
     RUNTIME_PROTOCOL_DECOS,
+    FakeExpression,
 )
 from mypy.tvar_scope import TypeVarScope
 from mypy.typevars import fill_typevars
@@ -1260,7 +1261,7 @@ class SemanticAnalyzerPass2(NodeVisitor[None],
         if defn.fullname:
             hook = self.plugin.get_customize_class_mro_hook(defn.fullname)
             if hook:
-                hook(ClassDefContext(defn, Expression(), self))
+                hook(ClassDefContext(defn, FakeExpression(), self))
 
     def update_metaclass(self, defn: ClassDef) -> None:
         """Lookup for special metaclass declarations, and update defn fields accordingly.

--- a/test-data/unit/check-custom-plugin.test
+++ b/test-data/unit/check-custom-plugin.test
@@ -597,3 +597,10 @@ plugins=<ROOT>/test-data/unit/plugins/callable_instance.py
 [file mypy.ini]
 [[mypy]
 plugins=<ROOT>/test-data/unit/plugins/depshook.py
+
+[case testCustomizeMroTrivial]
+# flags: --config-file tmp/mypy.ini
+class A: pass
+[file mypy.ini]
+[[mypy]
+plugins=<ROOT>/test-data/unit/plugins/customize_mro.py

--- a/test-data/unit/plugins/customize_mro.py
+++ b/test-data/unit/plugins/customize_mro.py
@@ -1,0 +1,10 @@
+from mypy.plugin import Plugin
+
+class DummyPlugin(Plugin):
+    def get_customize_class_mro_hook(self, fullname):
+        def analyze(classdef_ctx):
+            pass
+        return analyze
+
+def plugin(version):
+    return DummyPlugin

